scripts/ingest_GWAS_Catalog.py

```diff
@@ -0,0 +1,288 @@
+# associations = 'https://www.ebi.ac.uk/gwas/api/search/downloads/alternative'  # URL to GWAS Catalog associations
+GWAS_CATALOG_ASSOCIATIONS = 'gs://ot-team/dsuveges/v2d_files/gwas_associations_2022-06-27.tsv'
+
+##
+## Temporary output files:
+##
+OUTPUT_PATH = 'gs://ot-team/dsuveges/v2d_files/'
+
+# These parameters will be read from a config file:
+
+
+# Ingesting GWAS Catalog associations:
+DROPPED_STUDIES = [
+    'GCST005806',  # Sun et al pQTL study
+    'GCST005837'  # Huang et al IBD study
+]
+
+# This flag will allow to run the code for PPP allowing the ingestion of unpublished studies:
+INGEST_UNPUBLISHED_STUDIES = False
+
+# P-value threshold for associations:
+PVALCUTOFF = 5e-8
+
+ASSOCIATION_COLUMNS_MAP = {
+    # Variant related columns:
+    'STRONGEST SNP-RISK ALLELE': 'strongest_snp_risk_allele',  # variant id and the allele is extracted (; separated list)
+    'CHR_ID': 'chr_id',  # Mapped genomic location of the variant (; separated list)
+    'CHR_POS': 'chr_pos',
+    'RISK ALLELE FREQUENCY': 'risk_allele_frequency',
+    'CNV': 'cnv',  # Flag if a variant is a copy number variant
+    'SNP_ID_CURRENT': 'snp_id_current',  #
+    'SNPS': 'snp_ids',  # List of all SNPs associated with the variant
+
+    # Study related columns:
+    'STUDY ACCESSION': 'study_accession',
+
+    # Disease/Trait related columns:
+    'DISEASE/TRAIT': 'disease_trait',  # Reported trait of the study
+    'MAPPED_TRAIT_URI': 'mapped_trait_uri',  # Mapped trait URIs of the study
+    'MERGED': 'merged',
+    'P-VALUE (TEXT)': 'p_value_text',  # Extra information on the association.
+
+    # Association details:
+    'P-VALUE': 'p_value',  # p-value of the association, string: split into exponent and mantissa.
+    'PVALUE_MLOG': 'pvalue_mlog',  # -log10(p-value) of the association, float
+    'OR or BETA': 'or_beta',  # Effect size of the association, Odds ratio or beta
+    '95% CI (TEXT)': 'confidence_interval',  # Confidence interval of the association, string: split into lower and upper bound.
+
+    'CONTEXT': 'context',
+}
+
+
+def read_GWAS_associations() -> DataFrame:
+    '''This function retrieves the GWAS Catalog association dataset'''
+
+    # Reading and filtering associations:
+    association_df = (
+        spark.read.csv(GWAS_CATALOG_ASSOCIATIONS, sep='\t', header=True)
+
+        # Select and rename columns:
+        .select(*[f.col(old_name).alias(new_name) for old_name, new_name in ASSOCIATION_COLUMNS_MAP.items()])
+
+        # Cast columns:
+        .withColumn('pvalue_mlog', f.col('pvalue_mlog').cast(t.FloatType()))
+
+        # Apply some pre-defined filters on the data:
+        .filter(
+            (~f.col('chr_id').contains(' x '))  # Dropping associations based on variant x variant interactions
+            & (f.col('pvalue_mlog') > -np.log10(PVALCUTOFF))  # Dropping sub-significant associations
+            & (f.col('chr_pos').isNotNull() & f.col('chr_id').isNotNull())  # Dropping associations without genomic location
```

Comment:
How many of these association roughly are without annotation for a position?

Reply:
So 18246 (4.5%) of all associations don't have chr:pos information. Out of which 2636 has rsID. We can jump all the hoops to recover them, which was actually happening in the old pipelines, but I think this is an overcomplication.

```python
df = pd.read_csv('https://www.ebi.ac.uk/gwas/api/search/downloads/full', sep='\t')
print(f'Number of associations without chromosome/position: {len(df.loc[df.CHR_POS.isna()])}')
print(f'Number of associations without chromosome/position but with rsid: {len(df.loc[(df.CHR_POS.isna())&(df.SNPS.str.startswith("rs"))])}')

```